pkg/foghorn/controller_test.go

```diff
@@ -0,0 +1,161 @@
+			BranchProtection:       config.BranchProtection{},
+			Orgs:                   nil,
+			Gerrit:                 config.Gerrit{},
+			JenkinsOperators:       nil,
+			LighthouseJobNamespace: "",
+			PodNamespace:           "",
+			LogLevel:               "",
+			PushGateway:            config.PushGateway{},
+			OwnersDirExcludes:      nil,
+			OwnersDirBlacklist:     nil,
+			PubSubSubscriptions:    nil,
+			GitHubOptions:          config.GitHubOptions{},
+			ProviderConfig: &config.ProviderConfig{
+				Kind:    "fake",
+				Server:  "https://github.com",
+				BotUser: "jenkins-x-bot",
+			},
+		},
+	})
+
+	pluginAgent := &plugins.ConfigAgent{}
+	pluginAgent.Set(&plugins.Configuration{
+		Plugins:                    nil,
+		ExternalPlugins:            nil,
+		Owners:                     plugins.Owners{},
+		Approve:                    nil,
+		UseDeprecatedSelfApprove:   false,
+		UseDeprecatedReviewApprove: false,
+		Blockades:                  nil,
+		Cat:                        plugins.Cat{},
+		CherryPickUnapproved:       plugins.CherryPickUnapproved{},
+		ConfigUpdater:              plugins.ConfigUpdater{},
+		Heart:                      plugins.Heart{},
+		Label:                      plugins.Label{},
+		Lgtm:                       nil,
+		RepoMilestone:              nil,
+		RequireMatchingLabel:       nil,
+		RequireSIG:                 plugins.RequireSIG{},
+		SigMention:                 plugins.SigMention{},
+		Size:                       plugins.Size{},
+		Triggers:                   nil,
+		Welcome:                    nil,
+	})
+
+	cfgMapWatcher := &watcher.ConfigMapWatcher{}
+
+	ns := "jx"
+
+	for _, tc := range testCases {
+		t.Run(tc, func(t *testing.T) {
+			testData := path.Join("test_data", tc)
+			_, err := os.Stat(testData)
+			assert.NoError(t, err)
+
+			observedJob, err := loadLighthouseJob(testData, "observed-lhjob.yml")
+			assert.NoError(t, err)
+
+			expectedJob, err := loadLighthouseJob(testData, "expected-lhjob.yml")
+			assert.NoError(t, err)
+
+			// create fake controller
+			scheme := runtime.NewScheme()
+			err = lighthousev1alpha1.AddToScheme(scheme)
+			assert.NoError(t, err)
+			c := fake.NewFakeClientWithScheme(scheme, observedJob)
+			reconciler, err := NewLighthouseJobReconcilerWithConfig(c, scheme, ns, cfgMapWatcher, configAgent, pluginAgent)
+			assert.NoError(t, err)
+
+			// invoke reconcile
+			_, err = reconciler.Reconcile(ctrl.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: ns,
+					Name:      observedJob.GetName(),
+				},
+			})
+			assert.NoError(t, err)
+
+			var jobList lighthousev1alpha1.LighthouseJobList
+			err = c.List(nil, &jobList, client.InNamespace(ns))
+			assert.NoError(t, err)
+			assert.Len(t, jobList.Items, 1)
+			// Ignore status.starttime since that's always going to be different
+			updatedJob := jobList.Items[0].DeepCopy()
+			updatedJob.Status.StartTime = metav1.Time{}
+			if d := cmp.Diff(expectedJob, updatedJob); d != "" {
```

Comment:
I am a bit struggling to understand/see what you are actually testing here. Given the test files I assume that you are mainly looking for a "status change". But given the size of the the test yaml files and that there is no easy see the difference, it is a bit hard what you see what changed. 
When I diff the files, in one case the state changes from 'pending' to 'running'. Would it not be better to just assert on this field?

Reply:
Yeah, let me actually shrink that to comparing the `Status` fields - I want to make sure things get changed there with the `status-change` test but not with the `no-status-change` test.